### DISCUSSION TITLE
Pooler Error ClientHandler: socket closed with reason

### DISCUSSION THREAD

**Discussion by SayonB** [supabase/supabase#23272] [Questions]
All connections to my database is failing with the error "SSL connection has been closed unexpectedly"

When I go to https://supabase.com/dashboard/project/[PROJECT_ID]/logs/pooler-logs I see many logs with

`ClientHandler: socket closed with reason {:shutdown, :terminate_received}, DbHandler {#PID<50015.13801.5652>, {:error, {:exit, {:timeout, {:partisan_gen_statem, :call, [#PID<50015.13801.5652>, :get_state_and_mode, 5000]}}}}}`

and 

`ClientHandler: socket closed with reason {:shutdown, :terminate_received}`

we've tried restarting supabase cloud, upgrading it, removing SSL enforcing. What else can we do?

**Comment by GaryAustin1** [comment]
Please report this to support.

**Reply by IgnacioVegaMx** [reply]
I already reported it

**Comment by IgnacioVegaMx** [comment]
If the same thing happens to me, connecting from pgAdmin4 and PHP, any ideas on this?

The most curious thing is that it only happens to me in the project that is in the Pro plan, I have two others and I can access those, is it a matter of the Pro plan projects?

**Reply by GaryAustin1** [reply]
I doubt it is Pro plan, but more likely a specific supavisor server if that is the issue.
Also if you support ipv6 you can use the direct port and that is working for at least one user as a non-pooler need work around.

**Comment by groksrc** [comment]
I am having the same problem. We have opened a ticket.

**Comment by GaryAustin1** [comment]
https://status.supabase.com/incidents/h5t5p6y0jxvc
![image](https://github.com/supabase/supabase/assets/54564956/d7015684-41bb-4aee-9424-2cfc28437d5c)

**Comment by AlanChen4** [comment]
I'm using the Supabase connection pooler and was able to resolve this by increasing the `Pool Size` (Settings > Database > Connection Pooling Configuration). Default value for me was 15, and stopped having this issue after increasing to 48.

**Reply by quantchain14** [reply]
fuck sake you saved me

**Reply by peterlendtech503** [reply]
you saved me here Alan!

**Reply by alexjfalcao** [reply]
One more saved by Alan. Thanks

**Reply by hc-12** [reply]
Cheers, @AlanChen4, you've been an absolute lifesaver!

Is the recent PostgreSQL 17 upgrade related to the sudden increase in required pool size?
